### DISCUSSION TITLE
How to Bulk Install Devices in Device Bays

### DISCUSSION THREAD

**Discussion by mahmoudkhalil2** [netbox-community/netbox#11576] [Q&A]
Hello, How we can bulk install devices in the device bays ineasted of this manual action in this screenshot.

in my case I have more than 20 Cisco UCS 5108 Chassis and each chassis has 8 servers installed on it, which mean we have around 160 servers installed in all the UCS chassis.

I can bulk import the 20 Chassis.
I can bulk import the 160 servers in device types as a child 
I can bulk import the 160 servers in device page
But to install device in the device bays, I need to make it manually for the 160 servers, so is their any way to bulk install the devices in the device bays.
![image](https://user-images.githubusercontent.com/121564738/214549029-51fbe1bb-3e2e-4478-a4b4-49384f5113be.png)

**Comment by mtinberg** [comment]
Looking at my Netbox 3.4 system, parent and device_bay are available fields for Device import if the Child Devices tab is selected instead of Racked Devices, so you may just be on an older version and need to upgrade.  If you can't assign the slot when importing the Device record using CSV or YAML then I'd make a small script to assign the devices to the slots using the REST API, using whatever scripting language you and your team are comfortable with.

 Field Options
Field       Required    Accessor    Description
name  —     —     Name
device_role             name  Assigned role
tenant      —     name  Assigned tenant
manufacturer            name  Device type manufacturer
device_type             model       Device type model
platform    —     name  Assigned platform
serial      —     —     Serial number
asset_tag   —     —     A unique tag used to identify this device
status            —     
Operational status
parent            name  Parent device
device_bay        name  Device bay in which this device is installed
virtual_chassis   —     name  Virtual chassis
vc_position       —     —     Virtual chassis position
vc_priority       —     —     Virtual chassis priority
cluster     —     name  Virtualization cluster
comments    —     —     Comments
tags  —     slug  Tag slugs separated by commas, encased with double quotes (e.g. "tag1,tag2,tag3")
id    —     —     Numeric ID of an existing object to update (if not creating a new object)


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: mahmoudkhalil2 ***@***.***>
Sent: Wednesday, January 25, 2023 5:14 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] How to Bulk Install Devices in Device Bays (Discussion #11576)


Hello, How we can bulk install devices in the device bays ineasted of this manual action in this screenshot.

in my case I have more than 20 Cisco UCS 5108 Chassis and each chassis has 8 servers installed on it, which mean we have around 160 servers installed in all the UCS chassis.

I can bulk import the 20 Chassis.
I can bulk import the 160 servers in device types as a child
I can bulk import the 160 servers in device page
But to install device in the device bays, I need to make it manually for the 160 servers, so is their any way to bulk install the devices in the device bays.
[image]<https://user-images.githubusercontent.com/121564738/214549029-51fbe1bb-3e2e-4478-a4b4-49384f5113be.png>

**Comment by candlerb** [comment]
> (1) I can bulk import the 20 Chassis.
> (2) I can bulk import the 160 servers in device types as a child
> (3) I can bulk import the 160 servers in device page

I don't know what you mean by step (2).  There should be a single device type for the blade - or one for each variant of the blade, if you have a mixture of models.

Therefore, all you need is step (3), to import the servers (blades).

As Mark says: in Netbox 3.4, there is a separate tab under "device bulk import" for importing child devices: it takes you to `/dcim/devices/import/child-devices/`. This has **parent** and **device_bay** as (mandatory) CSV columns.
